### ISSUE TITLE
Bind conflicting with device setup

### ISSUE THREAD

**Issue by tkrzepka** [hyprwm/Hyprland#6276]
### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit fe7b748eb668136dd0558b7c8279bfcd7ab4d759  (props: bump version to 0.39.1).
Date: Tue Apr 16 16:01:03 2024
Tag: v0.39.1, commits: 4460

flags: (if any)


System Information:
System name: Linux
Node name: tomek-82a3
Release: 6.6.30-2-MANJARO
Version: #1 SMP PREEMPT_DYNAMIC Wed May  8 17:46:43 UTC 2024


GPU information:
00:02.0 VGA compatible controller [0300]: Intel Corporation TigerLake-LP GT2 [Iris Xe Graphics] [8086:9a49] (rev 01) (prog-if 00 [VGA controller])


os-release: NAME="Manjaro Linux"
PRETTY_NAME="Manjaro Linux"
ID=manjaro
ID_LIKE=arch
BUILD_ID=rolling
ANSI_COLOR="32;1;24;144;200"
HOME_URL="https://manjaro.org/"
DOCUMENTATION_URL="https://wiki.manjaro.org/"
SUPPORT_URL="https://forum.manjaro.org/"
BUG_REPORT_URL="https://docs.manjaro.org/reporting-bugs/"
PRIVACY_POLICY_URL="https://manjaro.org/privacy-policy/"
LOGO=manjarolinux


plugins:
```

</details>

### Bug or Regression?

Bug

### Description

if you have defined two keyboard devices with different kb_variants one of them will have wrong bindings
one thing i found is in ``` xev --event keyboard ```
one keyboard shows 
```
KeyPress event, serial 290, synthetic NO, window 0x400001,
    root 0x4a5, subw 0x0, time 3713059, (536,721), root:(3740,763),
--> state 0x0, keycode 43 (keysym 0x64, d), same_screen YES,
    XLookupString gives 1 bytes: (64) "d"
    XmbLookupString gives 1 bytes: (64) "d"
    XFilterEvent returns: False
```
and the other keyboard shows for the matching key pressed
```
KeyPress event, serial 292, synthetic NO, window 0x400001,
    root 0x4a5, subw 0x0, time 3724267, (507,702), root:(3711,744),
--> state 0x0, keycode 40 (keysym 0x64, d), same_screen YES,
    XLookupString gives 1 bytes: (64) "d"
    XmbLookupString gives 1 bytes: (64) "d"
    XFilterEvent returns: False
```
and the matching result for the keycode of the first keyboard
```
KeyRelease event, serial 290, synthetic NO, window 0x400001,
    root 0x4a5, subw 0x0, time 3710789, (536,721), root:(3740,763),
--> state 0x0, keycode 43 (keysym 0x68, h), same_screen YES,
    XLookupString gives 1 bytes: (68) "h"
    XFilterEvent returns: False
```

in the end on keyboard 1 the binding works with ```key d``` and on the other with ```key h``` which makes the it really troublesome to use
TLDR: bind matches based on keycode and not keysym/modmask even on a bind defined on key and NOT keycode

``` hyprctl binds|grep -B4 -A5 "key: D"  ```   
                                                                                                                                                                      
```
bind
	modmask: 64
	submap:
	key: D
	keycode: 0
	catchall: false
	dispatcher: exec
	arg: ~/.config/hypr/scripts/rofi_launcher
```

### How to reproduce

```
device {
    name = at-translated-set-2-keyboard
    kb_layout = us
    kb_variant = dvorak
}

device {
    name = zsa-technology-labs-moonlander-mark-i
    kb_layout = us
    kb_variant = basic
}
```
one of the keyboards will not work correctly with bindings

can be tested using one keyboard by defining in input kb_variant different than later defined keyboard device kb_variant

```
input {
        kb_model =
        kb_layout = us
        kb_variant = basic
        kb_rules =
        kb_file =
        numlock_by_default = false
        repeat_rate = 25
        repeat_delay = 600
        sensitivity = 1.0
        accel_profile = adaptive
        force_no_accel = false
        left_handed = false
        scroll_method = 2fg
        scroll_button = 0
        scroll_button_lock = 0
        natural_scroll = false
        follow_mouse = 1
        mouse_refocus = true
        float_switch_override_focus = 1
        touchpad {
                disable_while_typing = true
                natural_scroll = true
                scroll_factor = 1.0
                middle_button_emulation = false
                tap_button_map =
                clickfinger_behavior = false
                tap-to-click = true
                drag_lock = false
                tap-and-drag = true
        }
        touchdevice {
                transform = 0
                output =
        }
        tablet {
                transform = 0
                output =
                region_position = 0 0
                region_size = 0 0
                relative_input = false
        }
}

device {
    name = at-translated-set-2-keyboard
    kb_layout = us
    kb_variant = dvorak
}
```

**Comment by tkrzepka** [comment]
### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch bugfix_keybindmanager at commit a148394fce2f02b56cf965a1dfa32f9c13149c0c dirty (1).
Date: Thu May 30 05:12:10 2024
Tag: v0.40.0-1-ga148394f, commits: 4607

flags: (if any)


System Information:
System name: Linux
Node name: tomek-82a3
Release: 6.6.30-2-MANJARO
Version: #1 SMP PREEMPT_DYNAMIC Wed May  8 17:46:43 UTC 2024


GPU information: 
00:02.0 VGA compatible controller [0300]: Intel Corporation TigerLake-LP GT2 [Iris Xe Graphics] [8086:9a49] (rev 01) (prog-if 00 [VGA controller])


os-release: NAME="Manjaro Linux"
PRETTY_NAME="Manjaro Linux"
ID=manjaro
ID_LIKE=arch
BUILD_ID=rolling
ANSI_COLOR="32;1;24;144;200"
HOME_URL="https://manjaro.org/"
DOCUMENTATION_URL="https://wiki.manjaro.org/"
SUPPORT_URL="https://forum.manjaro.org/"
BUG_REPORT_URL="https://docs.manjaro.org/reporting-bugs/"
PRIVACY_POLICY_URL="https://manjaro.org/privacy-policy/"
LOGO=manjarolinux


plugins:
```

</details>

### Description

I did some testing to help out. I've compiled ```0.40.0``` and verified that the issue is with handling different layouts itself and not the handling of keybindings
when pressing Q on ```working keyboard (us,basic)```
```
bool CKeybindManager::handleKeybinds(...) SPressedKeyWithMods {
    name: 
    keysym: 0x71
    keycode: 24
    mod: 0
    sent: 1
    submap:
} 
```
key press result from xev
```
KeyPress event, serial 88, synthetic NO, window 0x400001,
    root 0x4a3, subw 0x0, time 2548572, (108,226), root:(3955,773),
    state 0x0, keycode 24 (keysym 0x71, q), same_screen YES,
    XLookupString gives 1 bytes: (71) "q"
    XmbLookupString gives 1 bytes: (71) "q"
    XFilterEvent returns: False
```

when pressing Q on ```keyboard with changed layout(us,dvorak)```
```
bool CKeybindManager::handleKeybinds(...) SPressedKeyWithMods {
    name: 
    keysym: 0x78
    keycode: 53
    mod: 0
    sent: 1
    submap:
} 
```
key press result from xev
```
KeyPress event, serial 88, synthetic NO, window 0x400001,
    root 0x4a3, subw 0x0, time 2595143, (243,284), root:(4090,831),
    state 0x0, keycode 53 (keysym 0x71, q), same_screen YES,
    XLookupString gives 1 bytes: (71) "q"
    XmbLookupString gives 1 bytes: (71) "q"
    XFilterEvent returns: False
```
notice that keysym 0x78 a letter x which in dvorak layout should be q(correctly detected in xev)

**Comment by vaxerski** [comment]
I might be reading this wrong, but isn't this _precisely_ what is defined in the wiki here 
![image](https://github.com/hyprwm/Hyprland/assets/43317083/9f9545f3-c9fc-4ecf-926b-5b3927328eb4)
https://wiki.hyprland.org/Configuring/Uncommon-tips--tricks/#switchable-keyboard-layouts

**Comment by tkrzepka** [comment]
huh
yup, my bad.
